### DISCUSSION TITLE
fcitx5 有无办法用数字键盘区输入英文标点？

### DISCUSSION THREAD

**Discussion by tankb52** [fcitx/fcitx5#335] [General]
就是句号变成小数点。
有时需要同时输入中文和带小数的数字，比如“价格￥4.30元”。
fcitx5-rime 是可以这样做的。
但是 fcitx5-rime 有个问题就是在 vim 中无法自动切换英文；之前在 firefox 的 bing 搜索结果界面也不能快速回删，所以才用了 fcitx5。 

不知道能不能设置成我想要的结果？

**Comment by xingwozhonghua126** [comment]
https://github.com/fcitx/fcitx5-chinese-addons

可以试试自定义标点功能

**Reply by wengxt** [reply]
这个其实不行因为目前这个映射是基于字符的，但是小键盘的按键其实是不会走标点相关的代码，判断在这里，小键盘应该直接走直接输入的。

https://github.com/fcitx/fcitx5-chinese-addons/blob/37a4e1a8e2aea764122ace8dd4d333c5b252b568/im/pinyin/pinyin.cpp#L1328

**Comment by wengxt** [comment]
现在拼音就是小键盘不输入全角标点啊？

**Reply by tankb52** [reply]
> 现在拼音就是小键盘不输入全角标点啊？

我发现问题了，在这里的回复框里输入时是符合预期的。 —— 价格￥4.30元。
但是在 vim 中输入时就是有问题的。——价格￥4。30元。

**Reply by wengxt** [reply]
？我不知道你用的什么 VIM，但是我用 xdotool key KP_Decimal 发送的都能正常输入 .

**Reply by wengxt** [reply]
https://github.com/fcitx/fcitx5-chinese-addons/commit/9d7058ab7bc37ea56ddb2550681d294f3ec07c92 具体修改在这里，如果你用的是别的输入法的话，那你可能得找别的输入法来处理

**Reply by wengxt** [reply]
码表的话看起来有个判断条件的 bug，如果是码表的话得等下个版本修复。如果是之外的输入法的话，那就只能找对应的输入法了
